nexus/src/internal_api/http_entrypoints.rs

```diff
@@ -402,8 +402,7 @@ async fn cpapi_metrics_collect(
 ) -> Result<HttpResponseOk<ProducerResults>, HttpError> {
     let context = request_context.context();
     let producer_id = path_params.into_inner().producer_id;
-    let handler =
-        async { collect(&context.producer_registry, producer_id).await };
+    let handler = async { collect(&context.producer_registry, producer_id) };
```

Comment:
Yep: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c5775d69030ddffbefbfd23e677da1f7

Arguably clippy should be complaining about an unused async _here_ now, but apparently it does not, or maybe it realizes that `handler` is required to be async because it's passed to a function that expects a future?